GS: Make GSUtil constexpr functions header only.

TJnotJT opened the pull request:
### Description of Changes
Make some of the utility functions in GSUtil header-only.

### Rationale behind Changes
The functions couldn't be used as constexpr in other files because they were defined in the cpp file instead of header. Some of these functions could be used in e.g. templated functions in GSVertexTrace as compile-time constants.

### Suggested Testing Steps
Changes should be transparently equivalent so no testing required.

### Did you use AI to help find, test, or implement this issue or feature?
Yes, code autocompletion w/ Github copilot.

Mrlinkwii reviewed (approved):
PCSX2 still works

TJnotJT commented:
Thanks for the review. If possible could the GS Texture Cache and GS Software labels be removed? They are no longer relevant as the changes are now contained to the GSUtil class.

Did a dump run on about ~600 dumps and no issues.

TellowKrinkle commented:
Which ones are you actually interested in using constexpr outside of GSUtil?  The inability to e.g. use static function-local variables without them getting duplicated everywhere leads to the code being much harder to read, so I'd rather you not move any into the header that don't have to be.

TellowKrinkle commented:
I feel like the string-returning ones are the least likely to be needed constexpr
(And if you're going to move them, at least format them nicely like `GetPSMName` was before you moved it)

TJnotJT committed:
GS: Make some GSUtil constexpr functions header only.

The functions for getting primclass and vertex counts may be used in constexpr contexts in templated functions.

TJnotJT commented:
That's probably true. I changed everything back except for the exact functions that I will be using in templates.

I formatted one of the switches but the other requires fall through so I left it as it is.

Let me know if I should make any other changes.

lightningterror reviewed (approved):
Did a dump run on hw only, no regressions.
